spec/unit/configurer_spec.rb

```diff
@@ -858,6 +858,34 @@ def expects_neither_new_or_cached_catalog
     end
   end
 
+  describe "when converging the environment" do
+    let(:apple) { Puppet::Resource::Catalog.new(Puppet[:node_name_value], Puppet::Node::Environment.remote('apple')) }
+    let(:banana) { Puppet::Resource::Catalog.new(Puppet[:node_name_value], Puppet::Node::Environment.remote('banana')) }
+
+    before :each do
+      apple.add_resource(resource)
+      banana.add_resource(resource)
+    end
+
+    it "converges after multiple attempts" do
+      expect(Puppet::Resource::Catalog.indirection).to receive(:find).and_return(apple, banana, banana)
+
+      allow(Puppet).to receive(:notice)
+      expect(Puppet).to receive(:notice).with("Local environment: 'production' doesn't match server specified environment 'apple', restarting agent run with environment 'apple'")
+      expect(Puppet).to receive(:notice).with("Local environment: 'apple' doesn't match server specified environment 'banana', restarting agent run with environment 'banana'")
```

Comment:
We unfortunately don't print the environment in the happy path. I thought about adding it to the `Applied catalog in X seconds`, but sadly there is code out that that depends on that exact message: https://github.com/puppetlabs/puppetlabs-tagmail/blob/0952eb6840be61c8be5eae35e19e1130e704e4cc/lib/puppet/reports/tagmail.rb#L205-L206

The reason I know this is because it's the one message that is always printed at `notice` level, so it's the only thing that is printed when not running with `--verbose`.

Another option is to add a new `info` message when the environment converges... but I leaned against it, to increase the likelihood that this doesn't introduce unexpected changes (like module testing).